src/realm/sync/client.cpp

```diff
@@ -1775,15 +1901,16 @@ void SessionWrapper::report_progress()
 util::Future<std::string> SessionWrapper::send_test_command(std::string body)
 {
     if (!m_sess) {
-        return util::Future<std::string>::make_ready(
-            Status{ErrorCodes::RuntimeError, "session must be activated to send a test command"});
+        return Status{ErrorCodes::RuntimeError, "session must be activated to send a test command"};
```

Comment:
Don't we need a Future here?

Reply:
A future will be "[copy constructed](https://en.cppreference.com/w/cpp/language/copy_initialization)" in the return using the Status value. The `Status` copy constructor in the `Future` class calls `make_ready` using the status value provided.

Reply:
got it. just noticed the implicit constructor